src/linpred.jl

```diff
@@ -148,18 +153,22 @@ function delbeta!(p::DensePredChol{T,<:CholeskyPivoted}, r::Vector{T}) where T<:
     p
 end
 
-function delbeta!(p::DensePredChol{T,<:Cholesky}, r::Vector{T}, wt::Vector{T}) where T<:BlasReal
+function delbeta!(p::DensePredChol{T,<:Cholesky}, r::Vector{T}, wt::Vector{T}; allowrankdeficient::Bool = false) where T<:BlasReal
     scr = mul!(p.scratchm1, Diagonal(wt), p.X)
     cholesky!(Hermitian(mul!(cholfactors(p.chol), transpose(scr), p.X), :U))
     ldiv!(p.chol, mul!(p.delbeta, transpose(scr), r))
     p
 end
 
-function delbeta!(p::DensePredChol{T,<:CholeskyPivoted}, r::Vector{T}, wt::Vector{T}) where T<:BlasReal
+function delbeta!(p::DensePredChol{T,<:CholeskyPivoted}, r::Vector{T}, wt::Vector{T}; allowrankdeficient::Bool = false) where T<:BlasReal
```

Comment:
I see that it only applies when `DensePredChol[T,<:CholeskyPivoted}`... should be fine for now since it would be the one used when they all use `CholeskyPivoted`... Maybe have `DensePredChol` with a field for `allowrankdeficient` rather than having to pass it?